app/src/main/java/com/github/se/polyfit/ui/screen/CreatePostScreen.kt

```diff
@@ -0,0 +1,161 @@
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.heightIn
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Edit
+import androidx.compose.material3.HorizontalDivider
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextField
+import androidx.compose.material3.TextFieldDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.hilt.navigation.compose.hiltViewModel
+import com.github.se.polyfit.R
+import com.github.se.polyfit.model.meal.Meal
+import com.github.se.polyfit.model.post.Location
+import com.github.se.polyfit.ui.components.button.PrimaryButton
+import com.github.se.polyfit.ui.components.dialog.LocationPermissionDialog
+import com.github.se.polyfit.ui.components.scaffold.CenteredTopBar
+import com.github.se.polyfit.ui.components.selector.MealSelector
+import com.github.se.polyfit.ui.components.selector.PictureSelector
+import com.github.se.polyfit.ui.theme.PrimaryPink
+import com.github.se.polyfit.ui.theme.PrimaryPurple
+import com.github.se.polyfit.ui.theme.SecondaryGrey
+import com.github.se.polyfit.viewmodel.post.CreatePostViewModel
+
+@Composable
+fun CreatePostScreen(
+    navigateBack: () -> Unit = {},
+    navigateForward: () -> Unit = {},
+    postViewModel: CreatePostViewModel = hiltViewModel()
+) {
+  val context = LocalContext.current
+  val meals by postViewModel.meals.collectAsState()
+  var isPermissionDialogDisplay by remember { mutableStateOf(false) }
+  var selectedMeal by remember { mutableStateOf(Meal.default()) }
+  val postComplete by remember(selectedMeal) { derivedStateOf { selectedMeal.isComplete() } }
+
+  fun setPostMeal(meal: Meal) {
+    postViewModel.setPostData(meal = meal)
+  }
+
+  fun onApprove() {
+    isPermissionDialogDisplay = false
+    postViewModel.setPostLocation(Location.default()) // TODO: Include location when ready
+    postViewModel.setPost()
+    navigateForward()
+  }
```

Comment:
This is because the implementation of location is not ready. Once it is, we would only do setPost and navigateForward here, so since its temporary it is not worth making into the view model